lib/internal/Magento/Framework/Image.php

```diff
@@ -52,7 +52,11 @@ public function open()
             throw new \Exception("File '{$this->_fileName}' does not exists.");
         }
 
-        $this->_adapter->open($this->_fileName);
+        try {
+            $this->_adapter->open($this->_fileName);
+        } catch (\Exception $e) {
+            throw new \Exception("{$e->getMessage()}: {$this->_fileName}");
```

Comment:
Why not add such information directly in `adapter->open()` method?

Reply:
So it doesn't need to be added to both existing adapters and any future adapters.

Reply:
It's a bit odd to catch exception and throw immediately to add information which was passed to adapter method anyway. Please move this addition to adapter(s) as it's clearly his responsibility to throw such exception.

As always, please do commit amend and force push.

Reply:
@orlangur ready for another look.